src/lib.rs

```diff
@@ -1516,12 +1532,29 @@ impl<T: Trait> Module<T> {
         // Store labels
         <ThreadLabels<T>>::mutate(new_thread_id, |value| *value = labels.clone());
 
+        // Update category for added thread
+        match sticky_index {
```

Comment:
Implement it this way to avoid the rule like:
if sticky_index {
   sticky_index.unwrap()
.............
Then I forget side-effects from match and update.
I think proper implementation is like following:
if let Some(unwrapped_sticky_index) = sticky_index {
} else {
}